### PR TITLE
SONARPY-1986: The IPythonSensor should register to .ipynb files

### PR DESCRIPTION
**Pull request by joke1196**

### CODE REVIEW

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java [hunk 1]
```diff
@@ -67,10 +77,38 @@ public void execute(SensorContext context) {
...
+      scanner.execute(pythonFiles, context);
+    } else {
+      processNotebooksFiles(pythonFiles, context);
+    }
+  }
+
+  private void processNotebooksFiles(List<PythonInputFile> pythonFiles, SensorContext context) {
+    pythonFiles = parseNotebooks(pythonFiles);
+    // Disable caching for IPynb files for now
+    CacheContext cacheContext = CacheContextImpl.dummyCache();
```
Comment (ghislainpiot) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java:91]: We should have that documented

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonPlugin.java [hunk 2]
```diff
@@ -91,6 +93,22 @@ public void define(Context context) {
...
+        .multiValues(true)
+        .category(PYTHON_CATEGORY)
+        .subCategory(GENERAL)
+        .onQualifiers(Qualifiers.PROJECT)
+        .defaultValue("ipynb")
+        .build(),
+      IPynb.class,
+      IPynbProfile.class,
+      IPynbSensor.class,
+      IPynbRuleRepository.class);
```
Comment (ghislainpiot) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonPlugin.java:110]: I don't know if we should have that property, is the goal to have other extensions or to allow disabling the notebooks ?

**Review by ghislainpiot — APPROVED** [review]
LGTM! Is the CI failure expected since the other ticket is not merged in that branch ?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonPlugin.java (hunk 2 above)
Comment (joke1196) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonPlugin.java:110]: No, you are right I don't think we need this. I think it was to mirror the Python file properties.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java (hunk 1 above)
Comment (joke1196) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java:92]: Done in https://sonarsource.atlassian.net/browse/SONARPY-2020

**Pull request merged**